hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/transform/StreamSourceTransform.java

```diff
@@ -85,8 +86,11 @@ public void addToDag(Planner p) {
             String v1name = name();
             Vertex v1 = p.dag.newVertex(v1name, metaSupplierFn.apply(eventTimePolicy))
                              .localParallelism(localParallelism());
+            int localParallelism = localParallelism() != LOCAL_PARALLELISM_USE_DEFAULT
```

Comment:
can be simplified as `int localParallelism = v1.determineLocalParallelism(localParallelism());`